samples/basic/blink_led/src/main.c

```diff
@@ -38,12 +44,18 @@ void main(void)
 		return;
 	}
 
-	/* Adjust max_period depending pwm capabilities */
-	pwm_get_cycles_per_sec(pwm_dev, PWM_CHANNEL, &cycles);
-	/* in very worst case, PWM has a 8 bit resolution and count up to 256 */
-	period = max_period = 256 * USEC_PER_SEC / cycles;
-	min_period = max_period / 64;
+	/* In case the default MAX_PERIOD value cannot be set for some PWM
+	 * hardware, try to decrease the value until it fits, but no further
+	 * than to the value of MIN_PERIOD muliplied by four (to allow the
+	 * sample to actually show some blinking with changing frequency).
+	 */
+	max_period = MAX_PERIOD;
+	while (pwm_pin_set_usec(pwm_dev, PWM_CHANNEL, max_period, 0)
```

Comment:
Perhaps it would be better to use `max_period / 2` as the last parameter (as it is in the call of this function in the main loop)? Just in case some driver handles the case of a 0-cycle pulse differently. Any opinions?

Reply:
i think that it might be safer to use `max_period/2`. just in case if driver ignores period when pulse is fixed.

Reply:
Changed. Added also `U` suffixes, just for consistency with the rest of the source code.